csharp/src/Ice/InputStream.cs

```diff
@@ -102,9 +100,10 @@ internal int Pos
         // size.
         private int _minTotalSeqSize = 0;
 
-        private IceInternal.Buffer _buf;
+        public ReadOnlyMemory<byte> Buffer => new ReadOnlyMemory<byte>(_buffer.Array);
 
-        private byte[]? _stringBytes; // Reusable array for reading strings.
+        private ArraySegment<byte> _buffer;
+        private int _tail;
```

Comment:
For InputStream, I would use `_pos` instead of `_tail`, since it can be anywhere in the buffer, not just at the end.

Reply:
fixed